src/tox/venv.py

```diff
@@ -514,7 +518,12 @@ def _pcall(
         os.environ.pop("VIRTUALENV_PYTHON", None)
         env = self._get_os_environ(is_test_command=is_test_command)
         bin_dir = str(self.envconfig.envbindir)
-        env["PATH"] = os.pathsep.join([bin_dir, os.environ["PATH"]])
+        if "PATH" in self.envconfig.setenv:
+            env["PATH"] = os.pathsep.join([bin_dir, self.envconfig.setenv["PATH"]])
+        else:
+            env["PATH"] = os.pathsep.join([bin_dir, os.environ["PATH"]])
+        # Remove redundantly duplicated directories in PATH without changing order
+        env["PATH"] = ":".join(list(dict.fromkeys(env["PATH"].split(":"))))
```

Comment:
this is (for instance) incorrect on windows -- use `os.pathsep` instead of `':'` -- also it's unnecessary (and imo a surprising behaviour) to remove duplicates so this modification probably shouldn't happen at all

Reply:
(above is true for `.split(":")` as well -- quite surprised this patch passes tests on windows?

Reply:
This is not passing on the windows test cases, and you gave me a solution. Thankyou.